lib/host-resolve.c

```diff
@@ -394,6 +394,15 @@ host_resolve_options_global_defaults(HostResolveOptions *options)
 static void
 _init_options(HostResolveOptions *options)
 {
+  if (options->use_dns == 0)
+    {
+      if (options->use_dns_cache != 0)
+        {
+          msg_warning("With use-dns(no), use-dns-cache() will be forced to 'no' too!",
+                      NULL);
+        }
+      options->use_dns_cache = 0;
+    }
```

Comment:
I am wondering if we want to do this conditional to the version number, e.g. if @version is less than 3.15 we would use the old defaults.

Also, the log message does not need punctuation at the end, syslog-ng automatically adds a ';'. With warnings we usually have a "WARNING:" prefix, especially when upgrading to a newer version.

And, the msg_warning() macros don't need a closing NULL anymore.

I think that's it.

Reply:
I suppose we can use `cfg_check_current_config_version()`, yeah. Will add that and adjust the patch based on your comments. Thanks!

Reply:
Hrm, the problem is, the place where we do this check, does not have access to the `GlobalConfig`... not sure it is worth the trouble to push it down there, or to implement the check higher up.

For the time being, I'll push an update that fixes the other issues.